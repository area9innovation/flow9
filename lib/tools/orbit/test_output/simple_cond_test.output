Testing simple conditional pattern matching
n = 5
n is positive
Now testing match statement
n is greater than zero
Result: 0
