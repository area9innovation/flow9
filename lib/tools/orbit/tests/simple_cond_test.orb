// Simple test for conditional pattern matching

println("Testing simple conditional pattern matching");

// Test with a single if
let n = 5;

// Print the value of n to debug
println("n = " + n);

// Note: We need explicit semicolons after each statement
if n > 0 then (
  println("n is positive")
) else (
  println("n is not positive")
);

println("Now testing match statement");

// Try a simple match with one condition
n is (
  x => (
    println("n is greater than zero")
  ) if x > 0;
  
  x => (
    println("n is not greater than zero")
  )
)