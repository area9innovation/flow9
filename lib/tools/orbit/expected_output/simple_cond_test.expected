Testing simple conditional pattern matching
n = 5
n is positive
Result: "n is positive"
